packages/mu-transport/index.js

```diff
@@ -39,17 +40,47 @@ module.exports = function transport (createDriver, mu, opts) {
     tearDown: driver.tearDown
   }
 
+
+
+  function constructFailResponse (err, msg) {
+    var pkt = { err: err,
+                response: {},
+                protocol: {}}
+    var elt = muid
+
+    /* istanbul ignore else  */
+    if (msg.protocol && msg.protocol.path && msg.protocol.path.length && msg.protocol.path.length > 0) {
+      elt = msg.protocol.path[msg.protocol.path.length - 1]
+      pkt.protocol.path = msg.protocol.path.slice(0, msg.protocol.path.length - 1)
+    }
+
+    /* istanbul ignore else  */
+    if (msg.protocol && msg.protocol.trace) {
+      pkt.protocol.trace = cloneDeep(msg.protocol.trace)
+    }
+
+    pkt.protocol.dst = elt
+    pkt.protocol.src = elt
+    pkt.protocol.inboundIfc = elt
+    return pkt
+  }
+
+
+
   function receive (err, msg) {
     logger.debug({in: msg}, 'message received')
     if (err) {
 
-      // received an error condition from the driver,
-      // typically this signals a failed client connection
-      // or other inbound connection error condition.
-      // In this case, log the error but make no attempt at
-      // further routing
+      // received an error condition from the driver, typically this signals a failed client connection
+      // or other inbound connection error condition. In this case, log the error but make no attempt at
+      // further routing unless we have an associated msg - the associated message will be used to construct
+      // an error response if present
       logger.error(err)
-      return
+      if (!msg || !msg.protocol) {
+        return
+      } else {
```

Comment:
I re-read...  I'm seeing a return and then an else - the else is redundant - what am I missing?